src/lib/containers/entity_finder/details/EntityDetailsList.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react'
+import useDeepCompareEffect from 'use-deep-compare-effect'
+import {
+  EntityHeader,
+  EntityType,
+  ProjectHeader,
+  Reference,
+} from '../../../utils/synapseTypes'
+import { GetProjectsParameters } from '../../../utils/synapseTypes/GetProjectsParams'
+import { SearchQuery } from '../../../utils/synapseTypes/Search'
+import { EntityChildrenDetails } from './configurations/EntityChildrenDetails'
+import { EntityHeaderListDetails } from './configurations/EntityHeaderListDetails'
+import { FavoritesDetails } from './configurations/FavoritesDetails'
+import { ProjectListDetails } from './configurations/ProjectListDetails'
+import { SearchDetails } from './configurations/SearchDetails'
+
+export enum EntityDetailsListDataConfigurationType {
+  HEADER_LIST, // simply displays one or more entity headers. incompatible with pagination
+  PARENT_CONTAINER, // retrieve the children for a given entity ID
+  USER_PROJECTS, // retrieve the user's projects (using the parameters)
+  USER_FAVORITES, // retrieve the user's favorites (using the parameters)
+  ENTITY_SEARCH, // Search for an entity
+}
+
+export type EntityDetailsListDataConfiguration = {
+  type: EntityDetailsListDataConfigurationType
+  /** Defined if type is HEADER_LIST */
+  headerList?: (EntityHeader | ProjectHeader)[]
+  /** Defined if type is PARENT_CONTAINER */
+  parentContainerParams?: {
+    parentContainerId: string
+  }
+  /** Defined if type is USER_PROJECTS */
+  getProjectParams?: GetProjectsParameters
+  /** Defined if type is ENTITY_SEARCH */
```

Comment:
another option is to define a configuration for each type, and then define EntityDetailsListDataConfiguration as the union of those type definitions